be/src/exec/vectorized/hdfs_scanner_text.cpp

```diff
@@ -49,66 +49,52 @@ class HdfsScannerCSVReader : public CSVReader {
     bool _should_stop_scan = false;
 };
 
-void HdfsScannerCSVReader::reset(size_t offset, size_t remain_length) {
-    _file->seek(offset);
+Status HdfsScannerCSVReader::reset(size_t offset, size_t remain_length) {
+    RETURN_IF_ERROR(_file->seek(offset));
     _offset = offset;
     _remain_length = remain_length;
     _should_stop_scan = false;
     _buff.skip(_buff.limit() - _buff.position());
+    return Status::OK();
 }
 
 Status HdfsScannerCSVReader::next_record(Record* record) {
-    if (_should_stop_scan) {
-        return Status::EndOfFile("Should stop for this reader!");
+    if (_remain_length == 0) {
```

Comment:
if the scan_range->length is happen to be in the position before `row_delimiter`, this might lost the first record of next scan_range

Reply:
No. Suppose remain length is 600, and delimiter is at 610.
- in the first round, remain_length != 0 , so it will read the record
- remain_length is updated to 0
- int he next round, it will return EOF.

Reply:
The old version has bug when the total csv file is small, it will only return the first record. I can reproduce this bug in my UT, and that's why I want to change/simplify this piece of code.

In my opinion, the old version `fill_buffer` mixed up 
- read data job
- and check if we can return EOF.

Instead we should:
- read data as much as possible, we don't check if we are at EOF position.
- we check if we are at EOF position in `next_record`.

Reply:
Yeah, it's my mistake. If scan range ends at row delimiter, one record will be skipped.